app/lib/s3_util.rb

```diff
@@ -0,0 +1,36 @@
+require "aws-sdk-s3"
+
+module S3Util
+  def self.s3_select_json(bucket, key, expression)
+    s3_select_params = {
+      bucket: bucket,
+      key: key,
+      expression_type: "SQL",
+      expression: expression,
+      input_serialization: {
+        json: {
+          type: "DOCUMENT",
+        },
+      },
+      output_serialization: {
+        json: {
+          record_delimiter: ",",
+        },
+      },
+    }
+
+    entry = []
+    begin
+      S3_CLIENT.select_object_content(s3_select_params) do |stream|
+        stream.on_records_event do |event|
+          entry.push(event.payload.read)
+        end
+      end
+    rescue Aws::S3::Errors => e
+      Rails.logger.error("Error retrieving entry #{expression} from #{bucket}/#{key} from s3")
+      Rails.logger.error(e.message)
+      return []
```

Comment:
Should this return an empty array, or an empty string? (it looks entry.join below is a string)

Reply:
You're right, fixed